snuba_sdk/entity.py

```diff
@@ -16,27 +16,38 @@ class InvalidEntity(Exception):
 class Entity(Expression):
     name: str
     alias: Optional[str] = None
-    sample: Optional[Union[int, float]] = None
+    sample: Optional[float] = None
 
     def validate(self) -> None:
         # TODO: There should be a whitelist of entity names at some point
         if not isinstance(self.name, str) or not entity_name_re.match(self.name):
             raise InvalidEntity(f"'{self.name}' is not a valid entity name")
 
         if self.sample is not None:
-            if not isinstance(self.sample, (int, float)):
-                raise InvalidEntity(
-                    "sample must be a float between 0 and 1 or an integer greater than 1"
-                )
-            elif isinstance(self.sample, float):
-                if self.sample < 0.0 or self.sample > 1.0:
-                    raise InvalidEntity(
-                        "float samples must be between 0.0 and 1.0 (%age of rows)"
-                    )
-            elif isinstance(self.sample, int):
-                if self.sample < 1:
-                    raise InvalidEntity("int samples must be at least 1 (# of rows)")
+            if not isinstance(self.sample, float):
+                raise InvalidEntity("sample must be a float")
+            elif self.sample <= 0.0:
+                raise InvalidEntity("samples must be greater than 0.0")
 
         if self.alias is not None:
             if not isinstance(self.alias, str) or not self.alias:
                 raise InvalidEntity(f"'{self.alias}' is not a valid alias")
+
+
+ENTITY_TIME_COLUMNS = {
```

Comment:
Ultimately this should be handled by the users of the SDK, not the SDK itself. For now this is here.